### PR TITLE
Adding kernels for Resize op

### PR DESCRIPTION
**Pull request by duli2012**

### CODE REVIEW

File: onnxruntime/core/providers/cpu/tensor/upsample.h [hunk 1]
```diff
@@ -56,8 +57,10 @@ class UpsampleBase {
   }
 
   void ScalesValidation(const std::vector<float>& scales, const UpsampleMode mode) const {
-    for (auto& scale : scales) {
-      ORT_ENFORCE(scale >= 1, "Scale value should be greater than or equal to 1.");
+    if (!is_resize) {
```
Comment (zhijxu-MS) [inline onnxruntime/core/providers/cpu/tensor/upsample.h]: if(is_resize){
//**make sure scale larger than 0**
.....
}

File: onnxruntime/test/providers/cpu/tensor/resize_op_test.cc [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+#include "core/providers/cpu/tensor/resize.h"
+#include "gtest/gtest.h"
+#include "test/providers/provider_test_utils.h"
+
+namespace onnxruntime {
+namespace test {
+
+TEST(ResizeOpTest, ResizeOpNearestTest) {
+  OpTester test("Resize", 10);
+  std::vector<float> scales{1.0f, 1.0f, 0.6f, 0.6f};
```
Comment (zhijxu-MS) [inline onnxruntime/test/providers/cpu/tensor/resize_op_test.cc]: maybe we should consider the case when scales is larger than 1?

File: onnxruntime/test/providers/cpu/tensor/resize_op_test.cc [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+  test.AddAttribute("mode", "nearest");
+
+  const int64_t N = 1, C = 1, H = 2, W = 4;
+  std::vector<float> X = {
+      1.0f, 2.0f, 3.0f, 4.0f,
+      5.0f, 6.0f, 7.0f, 8.0f};
+
+  test.AddInput<float>("X", {N, C, H, W}, X);
```
Comment (zhijxu-MS) [inline onnxruntime/test/providers/cpu/tensor/resize_op_test.cc]: add more test cases to cover other dtype such int?

File: onnxruntime/test/providers/cpu/tensor/resize_op_test.cc (hunk 2 above)
Comment (duli2012) [reply onnxruntime/test/providers/cpu/tensor/resize_op_test.cc]: that case was covered by previous upsample test cases.

File: onnxruntime/test/providers/cpu/tensor/resize_op_test.cc [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+      1.0f, 2.0f, 3.0f, 3.0f,
+      3.0f, 4.0f, 5.0f, 5.0f,
+
+      3.0f, 4.0f, 5.0f, 5.0f,
+      7.0f, 8.0f, 9.0f, 9.0f};
+
+  test.AddOutput<float>("Y", {N, C, (int64_t)(H * scales[2]), (int64_t)(W * scales[3])}, Y);
+  test.Run();
+}
+
```
Comment (hariharans29) [inline onnxruntime/test/providers/cpu/tensor/resize_op_test.cc]: LGTM.

Optional - Not sure if there is immediate use-case, but with the more "generic" resize spec - it's possible that this kernel be used in such a way that some scale values are > 1 and some < 1, effectively a "mixed" sampling (example - an image whose height is expanded while it's width is shrunk). If both of the current modes in the current implementation can handle - maybe it's worthwhile to check-in a test case (purely optional)

Comment (duli2012) [reply onnxruntime/test/providers/cpu/tensor/resize_op_test.cc]: Good point. Current implementation handles this case. Thank you for raising it.

**Comment by fdwr** [comment]
🤔 Do any known models use `int32` for resampling? I'm looking into the use case of `int32` (`uint8` and `float` usage are very clear) for WebNN in Chromium (which then calls OnnxRuntime.dll) to justify adding int32 or not. Thanks for any historical info.

[Context](https://chromium-review.googlesource.com/c/chromium/src/+/6701775/6/services/webnn/ort/context_impl_ort.cc#196):

> Dwayne R: For int32, does ORT really register it? (that seems weird 🤔)

> Shiyi Z: Yes it's registered [here](https://github.com/microsoft/onnxruntime/blob/main/onnxruntime/core/providers/cpu/cpu_execution_provider.cc#L1156), don't know about the use case though.

**Pull request merged**